tests/unit/client-eviction.tcl

```diff
@@ -479,8 +479,8 @@ start_server {} {
         assert {$total_client_mem <= $maxmemory_clients}
 
         # Make sure we have only half of our clients now
-        wait_for_condition 200 100 {
-            [llength [regexp -all -inline {name=client} [r client list]]] == $client_count / 2
+        wait_for_condition 20 100 {
+            [llength [regexp -all -inline {name=client} [r client list]]] <= $client_count / 2
```

Comment:
why change this?

Reply:
Perhaps we can determine whether running with io-thread. If it is, we will use the new verification and then add a TODO in the subsequent PR to fix it.
like
```
        wait_for_condition 200 100 {
            ([lindex [r config get io-threads] 1] == 1) ?
                ([llength [regexp -all -inline {name=client} [r client list]]] <= $client_count / 2) :
                ([llength [regexp -all -inline {name=client} [r client list]]] == $client_count / 2)
        } else {
            fail "Failed to evict clients"
        }
```

Reply:
Done.